multiqc/modules/hicpro/hicpro.py

```diff
@@ -0,0 +1,386 @@
+             'modify': lambda x: x * config.read_count_multiplier,
+             'shared_key': 'read_count',
+             'hidden': True
+         }
+         
+         headers['percent_valid'] = {
+             'title': '% Valid Pairs',
+             'description': 'Percentage of valid pairs over reported ones (%)',
+             'max': 100,
+             'min': 0,
+             'suffix': '%',
+             'scale': 'YlGn',
+             'hidden': True
+         }
+         
+         headers['valid_interaction_rmdup'] = {
+             'title': '{} Valid Pairs Unique'.format(config.read_count_prefix),
+             'description': 'Number of valid pairs after duplicates removal ({})'.format(config.read_count_desc),
+             'min': 0,
+             'scale': 'RdYlBu',
+             'modify': lambda x: x * config.read_count_multiplier,
+             'shared_key': 'read_count',
+             'hidden': True
+         }
+         
+         headers['percent_valid_interaction_rmdup'] = {
+             'title': '% Valid Pairs Unique',
+             'description': 'Percent of valid pairs over reported ones after duplicates removal (%)',
+             'max': 100,
+             'min': 0,
+             'suffix': '%',
+             'scale': 'YlGn',
+             'hidden': False
+         }
+     
+         self.general_stats_addcols(self.hicpro_data, headers, 'HiC-Pro')
+
+
+    def hicpro_mapping_chart (self):
+        """ Generate the HiC-Pro Aligned reads plot """
+
+        # Specify the order of the different possible categories
+        keys = [OrderedDict(), OrderedDict()]
+        keys[0]['Full_Alignments_Read']   = { 'color': '#005ce6', 'name': 'Full reads Alignments' }
+        keys[0]['Trimmed_Alignments_Read'] = { 'color': '#3385ff', 'name': 'Trimmed reads Alignments' }
+        keys[0]['Failed_To_Align_Read']     = { 'color': '#a9a2a2', 'name': 'Failed To Align' }
+        keys[1]['Full_Alignments_Read']   = { 'color': '#005ce6', 'name': 'Full reads Alignments' }
+        keys[1]['Trimmed_Alignments_Read'] = { 'color': '#3385ff', 'name': 'Trimmed reads Alignments' }
+        keys[1]['Failed_To_Align_Read']     = { 'color': '#a9a2a2', 'name': 'Failed To Align' }
+                           
+        data = [{},{}]
+        for s_name in self.hicpro_data:
+            for r in [1,2]:
+                data[r-1][s_name] = {
+                    'Full_Alignments_Read': self.hicpro_data[s_name]['global_R{}'.format(r)],
+                    'Trimmed_Alignments_Read': self.hicpro_data[s_name]['local_R{}'.format(r)],
+                    'Failed_To_Align_Read': int(self.hicpro_data[s_name]['total_R{}'.format(r)]) - int(self.hicpro_data[s_name]['mapped_R{}'.format(r)])
+                }
+        
+        # Config for the plot
+        config = {
+            'id': 'hicpro_mapping_stats_plot',
+            'title': 'HiC-Pro: Mapping Statistics',
+            'ylab': '# Reads',
+            'data_labels': ['Read 1', 'Read 2']
```

Comment:
Could be nice to specify the y axis labels too, again for clearer stand-alone plots:

```python
'ylab': '# Reads: Read 1',
'data_labels': [
        {'name': 'Read 1', 'ylab': '# Reads: Read 1'},
        {'name': 'Read 2', 'ylab': '# Reads: Read 2'}
]
```

Reply:
thanks